tests/test_inference.py

```diff
@@ -54,6 +55,77 @@ def assertTreeSequencesEqual(self, ts1, ts2):
         self.assertEqual(t1.mutations, t2.mutations)
 
 
+class TestUnfinalisedErrors(unittest.TestCase):
+    def test_generate_ancestors(self):
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            for path in [None, tmpfile.name]:
+                sample_data = tsinfer.SampleData(path=path, sequence_length=2)
+                sample_data.add_site(1, genotypes=[0, 1, 1, 0], alleles=["G", "C"])
+                print(dir(sample_data))
+                self.assertRaises(ValueError, tsinfer.generate_ancestors, sample_data)
+                sample_data.finalise()
+                tsinfer.generate_ancestors(sample_data)
+
+    def test_make_ancestor_data(self):
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            for path in [None, tmpfile.name]:
+                sample_data = tsinfer.SampleData(path=path, sequence_length=2)
+                sample_data.add_site(1, genotypes=[0, 1, 1, 0], alleles=["G", "C"])
+                self.assertRaises(ValueError, tsinfer.AncestorData, sample_data)
+
+    def test_match_ancestors_unfinalised_samples(self):
+        with tempfile.NamedTemporaryFile() as tmpfile:
```

Comment:
I don't think we need to repeat the check on "with a file or not" for each test. We can probably just do this once each for samples files and ancestors files.

Reply:
OK